docs/book/src/providers/v1alpha2-to-v1alpha3.md

```diff
@@ -83,3 +83,12 @@
 
 - Cosmetic change to set Cluster `Status.Phase` to `Provisioned` after at least one APIEndpoint is available. Previously only InfrastructureReady was being used to transition to `Provisioned`, this change now requires at least one APIEndpoint as well.
 - See https://github.com/kubernetes-sigs/cluster-api/pull/1721/files.
+
+## `Status.ErrorReason` and `Status.ErrorMessage` fields, populated to signal a fatal error has occurred, have been renamed in Cluster, Machine and MachineSet
+
+-  `Status.ErrorReason` has been renamed to `Status.FailureReason`
+-  `Status.ErrorMessage` has been renamed to `Status.FailureMessage`
+
+## The `external.ErrorsFrom` function has been renamed to `external.FailuresFrom`
+
+- The function has been modified to reflect the rename of `Status.ErrorReason` to `Status.FailureReason` and `Status.ErrorMessage` to `Status.FailureMessage`.
```

Comment:
We also need to state that now the external objects will need to rename as well, given that we're now looking in a different place. For example, AWSCluster and AWSMachine have to make this change as well.